src/app/navbar/navbar.component.ts

```diff
@@ -1,11 +1,21 @@
 import { Component } from '@angular/core';
+import { OAuthService } from 'angular-oauth2-oidc';
+
+import { environment } from '../../environments/environment';
+
 
 @Component({
   selector: 'app-navbar',
   templateUrl: './navbar.component.html',
   styleUrls: ['./navbar.component.css']
 })
 export class NavbarComponent {
+  showLogout: boolean = environment.enableAuth;
+
+  constructor(private auth: OAuthService) { }
 
-  constructor() { }
+  logout(event: Event): void {
+    this.auth.logOut();
+    event.preventDefault();
```

Comment:
Maybe a comment about why this preventDefault() call is necessary would be helpful as well

Reply:
Ah, the preventDefault is just to prevent the default behavior of clicking a link. Basically, since the link just executes JS and the JS handles the redirect, we don't want the link to execute it's default behavior of going to the value in the `href` attribute, which is just the convention `#`.

Reply:
Ah ok that makes sense.